jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -631,7 +635,9 @@ private String setupSettingFile(@Nonnull Collection<Credentials> credentials) th
 
         // Settings from Config File Provider
         if (StringUtils.isNotEmpty(step.getMavenSettingsConfig())) {
-            console.format("[withMaven] use Maven settings provided by the Jenkins Managed Configuration File '%s' %n", step.getMavenSettingsConfig());
+            if (LOGGER.isLoggable(Level.FINE)) {
+                console.format("[withMaven] using Maven settings provided by the Jenkins Managed Configuration File '%s' %n", step.getMavenSettingsConfig());
+            }

```

Comment:
:+1: but unrelated and could have been a separate PR? :-). More seriously, this makes it indeed harder to wrap one's head around the changes here about supporting settings per folder, so longer reviews, or no reviews at all if people don't reach the end of it...

Reply:
it is just an use -> using does it really make it harder?

Ohhhh you mean the Level.FINE, it was a feedback in the PR, but yeah I could have created another PR just for that.

Reply:
Note: an important goal is, for Level.INFO, to have one unique message in the build logs indicating which Maven settings is used and one for which maven global settings is used.
We have been having too verbose outputs at Level.INFO level so far.

Reply:
We do have only one with this change.